cider/exec/nextgen/jitlib/base/JITValueOperations.h

```diff
@@ -65,6 +65,42 @@ inline std::any castConstant(JITTypeTag target_type, T value) {
 }
 };  // namespace op_utils
 
+JITValuePointer operator&&(JITValue& lh, JITValue& rh) {
+  return lh.andOp(rh);
+}
+
+JITValuePointer operator&&(JITValue& lh, bool rh) {
+  auto& parent_func = lh.getParentJITFunction();
+  auto type = lh.getValueTypeTag();
+  JITValuePointer rh_pointer =
+      parent_func.createConstant(type, op_utils::castConstant(type, rh));
+  return lh && *rh_pointer;
+}
+
+JITValuePointer operator&&(bool lh, JITValue& rh) {
+  return rh && lh;
+}
+
+JITValuePointer operator||(JITValue& lh, JITValue& rh) {
+  return lh.orOp(rh);
+}
+
+JITValuePointer operator||(JITValue& lh, bool rh) {
+  auto& parent_func = lh.getParentJITFunction();
+  auto type = lh.getValueTypeTag();
+  JITValuePointer rh_pointer =
+      parent_func.createConstant(type, op_utils::castConstant(type, rh));
+  return lh || *rh_pointer;
+}
+
+JITValuePointer operator||(bool lh, JITValue& rh) {
+  return rh || lh;
```

Comment:
`return lh || rh;`

Reply:
> `return lh || rh;`

I'm afraid not, that would call itself recursively.

Reply:
> maybe `return operator||(rh, lh);` would be more clear.

Right, it works the same, could be more clear.